packages/vite/src/node/plugins/worker.ts

```diff
@@ -24,7 +24,11 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
     name: 'vite:worker',
 
     load(id) {
-      if (isBuild && parseWorkerRequest(id)?.worker != null) {
+      if (
+        isBuild &&
+        (parseWorkerRequest(id)?.worker ??
+          parseWorkerRequest(id)?.sharedworker) != null
```

Comment:
**nitpick (non-blocking):** I think the second optional chaining isn't needed, cause it's already checked in the first check
```suggestion
          parseWorkerRequest(id).sharedworker) != null
```

Reply:
The first optional chaining checks `worker` property of qs parsed search query. We still need to check for `sharedworker`, right? 🤔

Reply:
```ts
(parseWorkerRequest(id)?.worker ?? parseWorkerRequest(id).sharedworker) != null
// Is the same as
(parseWorkerRequest(id)?.worker ?? parseWorkerRequest(id)?.sharedworker) != null

// I think it becomes a bit clearer this way
const temp = parseWorkerRequest(id)
(temp?.worker ?? temp.sharedworker) != null
```

Reply:
Oh, you are right, I just made a mistake. I like your recommended style, but TS complains with single `?.`. TS won't know if because `temp` is nullish or `temp.worker` is nullish so it requires write `?.` again.

![image](https://user-images.githubusercontent.com/12322740/111911805-efa70400-8aa1-11eb-82db-d9e2df09c6f6.png)

I think we had to use double `?.` for type checking. Or checks `parsedQuery` in advance, but maybe a little verbose.

```ts
      if (
        isBuild &&
        (parsedQuery && (parsedQuery.worker ?? parsedQuery.sharedworker)) !=
          null
      ) {
        return ''
      }
```

Reply:
Ah yeah you are right (I was wrong I think)
Cause if `parsedQuery` would be `undefined`, then it falls into the second case and would be also `undefined` there

But I think your last example checking `parsedQuery` first could also be a solution

we loose short circuit here when calling the function before `isBuild` was checked
so maybe the best solution of both worlds would be to nest two `if`s

```ts
if (isBuild) {
  const parsedQuery = parseWorkerRequest(id)
  if (parsedQuery && (parsedQuery.worker || parsedQuery.sharedWorker)) {
// maybe:  if (parsedQuery && (!!parsedQuery.worker || !!parsedQuery.sharedWorker)) {
    return ''
  }
}
```

Reply:
Done

Reply:
Made a little change. 

```ts
parsedQuery.worker ?? parsedQuery.sharedworker
```
Should use `??` because the value will be `''` if the property exists.

```ts
!= null
```

Use `!= null` to distinguish between `''` and `undefined`